### PR TITLE
Avatar & Animation cppcheck static analysis fixes

### PR DESCRIPTION
**Pull request by hyperlogic**
Mostly stylistic, but It did detect several cases were we could have used uninitialized values.

Also, renamed JS properties, MyAvatar.FROM_HEAD, FROM_CAMERA & CUSTOM enumeration to MyAvatar.audioListenerModeHead, MyAvatar.audioListenerModeCamera & MyAvatar.audioListenerModeCustom

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1e94d9bdf58f04e65cc299fef0fdf37f2117b757 available at: https://gist.github.com/4ad1ca71b6f7cd14ca73<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7348.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7348.dmg

**Comment by OmegaHeron** [comment]
Tested Windows 7 - *Basic* smoke test -- didn't see anything failing to work, scripts I use work, av moves etc.

File: libraries/avatars/src/AvatarData.cpp [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ void AvatarData::detachAll(const QString& modelURL, const QString& jointName) {
         if (it->modelURL == modelURL && (jointName.isEmpty() || it->jointName == jointName)) {
             it = attachmentData.erase(it);
         } else {
-            it++;
+            ++it;
```
Comment (Atlante45) [inline libraries/avatars/src/AvatarData.cpp]: :+1:

File: interface/src/avatar/MyAvatar.h [hunk 2]
```diff
@@ -63,9 +63,9 @@ class MyAvatar : public Avatar {
     Q_PROPERTY(AudioListenerMode audioListenerMode READ getAudioListenerMode WRITE setAudioListenerMode)
     Q_PROPERTY(glm::vec3 customListenPosition READ getCustomListenPosition WRITE setCustomListenPosition)
     Q_PROPERTY(glm::quat customListenOrientation READ getCustomListenOrientation WRITE setCustomListenOrientation)
-    Q_PROPERTY(AudioListenerMode FROM_HEAD READ getAudioListenerModeHead)
-    Q_PROPERTY(AudioListenerMode FROM_CAMERA READ getAudioListenerModeCamera)
-    Q_PROPERTY(AudioListenerMode CUSTOM READ getAudioListenerModeCustom)
+    Q_PROPERTY(AudioListenerMode audioListenerModeHead READ getAudioListenerModeHead)
+    Q_PROPERTY(AudioListenerMode audioListenerModeCamera READ getAudioListenerModeCamera)
+    Q_PROPERTY(AudioListenerMode audioListenerModeCustom READ getAudioListenerModeCustom)
```
Comment (Atlante45) [inline interface/src/avatar/MyAvatar.h]: Seems pretty likely we have scripts using those.
We should check and update them accordingly.

Comment (Atlante45) [inline interface/src/avatar/MyAvatar.h]: Nvm, I see you've already done that.

**Pull request merged**